client/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable react/prop-types */
+
+import React from 'react';
+
+import Icon from './common/Icon';
+
+export interface Strings {
+  DASHBOARD: string;
+  EDIT_YOUR_ACCOUNT: string,
+  SEARCH: string,
+}
+
+export interface ModuleRenderContext {
+  slim: boolean;
+  expandingOrCollapsing: boolean;
+  currentPath: string;
+  strings: Strings;
+  navigate(url: string): Promise<void>;
+}
+
+export interface ModuleDefinition {
+  render(context: ModuleRenderContext): React.ReactFragment;
+}
+
+export interface SidebarProps {
+  modules: ModuleDefinition[];
+  currentPath: string;
+  strings: Strings;
+  navigate(url: string): Promise<void>;
+  onExpandCollapse?(collapsed: boolean);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const usePersistedState = <T, _>(key: string, defaultValue: T): [T, (value: T) => void]  => {
```

Comment:
Question: is this function used only in this component ?
It could be useful to have it defined in the `utils` directory https://github.com/wagtail/wagtail/tree/main/client/src/utils

Reply:
Yep, it is. I will probably remove this though in favour of storing the collapsed setting in the user's session on the backend. This is so that we can put the class in the Django template so that the initial render is at the correct width (currently, it's a bit janky for slim sidebar users)

Reply:
I've removed this snippet for now, will replace it with something on the backend in another PR